src/test/scala/org/apache/spark/sql/delta/MergeIntoScalaSuite.scala

```diff
@@ -19,14 +19,49 @@ package org.apache.spark.sql.delta
 import java.util.Locale
 
 import io.delta.tables._
-
 import org.apache.spark.sql._
+import org.apache.spark.sql.delta.sources.DeltaSQLConf
 import org.apache.spark.sql.types.StructType
 
 class MergeIntoScalaSuite extends MergeIntoSuiteBase {
 
   import testImplicits._
 
+  test(s"test merge optimizeMatchedOnlyMerge enabled:true update") {
```

Comment:
please add these to MergeIntoSuiteBase. We will add merge sql tests in the future using the base and we want these new tests to be automatically run on sql as well.